### PR TITLE
675 implement plugin that uses google translator

### PR DESCRIPTION
**Pull request by Mefpef**
Type pull request title with short description (#675 implement plugin that uses google translator )

----

**Please** following this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Give the same name to the pull request as the name of ticket 
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Test your code throughly before PR.
 
If your pull request is about ~20 lines of code add license information in your code in form of comment. i.e. # License: MIT, Author:  Your name.

To make clear that you license your contribution under the MIT you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [MIT license](https://opensource.org/licenses/MIT)

PRs without ticked MIT license will not be accepted - Sorry.

### CODE REVIEW

File: tracardi/process_engine/action/v1/google_translator_action.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from googletrans import Translator
+
+
+class Config(PluginConfig):
+    text_to_translate: str
+    source_language: str
+
+    @validator("source_language")
+    def check_given_source_language_type(cls, value):
+        if not isinstance(value, str) and value != "":
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/google_translator_action.py:19]: if not isinstance(value, str) or not value

File: tracardi/process_engine/action/v1/google_translator_action.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                    type="text",
+                                    props={
+                                        "label": "Text to translate"
+                                    }
+                                )
+                            ),
+                            FormField(
+                                id="source_language",
+                                name="Language source",
+                                description="Please provide source language of text that you want to translate.",
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/google_translator_action.py:77]: is it name of the language or something like "de", "en", etc? you should specify it.

File: tracardi/process_engine/action/v1/google_translator_action.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        metadata=MetaData(
+            name="Google translate",
+            desc="Translate the given text",
+            icon="global",
+            groups=["Operations"],
+            documentation=Documentation(
+                inputs={
+                    "payload": PortDoc(desc="This port takes payload object.")
+                },
+                outputs={"translation": PortDoc(desc="This port return translated text.")}
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/google_translator_action.py:99]: returns

File: tracardi/process_engine/action/v1/google_translator_action.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def register() -> Plugin:
+    return Plugin(
+        start=False,
+        spec=Spec(
+            module=__name__,
+            className="GoogleTranslateAction",
+            inputs=["payload"],
+            version="0.7.1",
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/google_translator_action.py:50]: 0.7.2

File: tracardi/process_engine/action/v1/google_translator_action.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+class GoogleTranslateAction(ActionRunner):
+    config: Config
+    translator: Translator
+
+    async def set_up(self, init):
+        self.config = validate(init)
+        self.translator = Translator()
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+        translation = self.translator.translate(self.config.text_to_translate, src=self.config.source_language,
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/google_translator_action.py:37]: well, this way you do not support dot paths. User can hard-code some text, but nothing more. Text should be accessed with DotAccessor instance.

File: tracardi/process_engine/action/v1/google_translator_action.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            manual="google_translate_action",
+            form=Form(
+                groups=[
+                    FormGroup(
+                        name="Google translator configurator",
+                        fields=[
+                            FormField(
+                                id="text_to_translate",
+                                name="Text to translate",
+                                description="Please provide text that you want to translate.",
```
Comment (DawidekZagajnik) [inline tracardi/process_engine/action/v1/google_translator_action.py:66]: path to text that you...

File: tracardi/service/setup/setup_plugins.py [hunk 7]
```diff
@@ -5,6 +5,11 @@
 
 installed_plugins: Dict[str, PluginTestTemplate] = {
 
+    "tracardi.process_engine.action.v1.google_translator_action": PluginTestTemplate(
+        init=None,
```
Comment (DawidekZagajnik) [inline tracardi/service/setup/setup_plugins.py:9]: init={"text_to_translate": "Hello", "source_language": "en"}

**Review by DawidekZagajnik — CHANGES_REQUESTED** [review]
needs some changes, check comments

File: tracardi/process_engine/action/v1/google_translator_action.py (hunk 6 above)
Comment (atompie) [reply tracardi/process_engine/action/v1/google_translator_action.py:66]: You can also name it a dot Notation reference or just reference to data conatining Text for translation.

**Review by atompie — APPROVED** [review]

**Pull request merged**